### PR TITLE
Sorting benchmark

### PR DESCRIPTION
**Pull request by frank-lsf**
Not included in this PR:

- Cluster setup scripts
- The Cython/C++ module for sort and merge
- Binaries (gensort, valsort, Prometheus setup)

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/experimental/raysort/main.py [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+import random
+import subprocess
+from typing import Iterable, List, Union
+
+import numpy as np
+import ray
+
+from ray.experimental.raysort import constants
+from ray.experimental.raysort import logging_utils
+from ray.experimental.raysort import sortlib
```
Comment (stephanie-wang) [inline python/ray/experimental/raysort/main.py:15]: Will this work without this dependency?

**Review by stephanie-wang — CHANGES_REQUESTED** [review]
Thanks, the main sort code looks good! Can you:
1. Add a simple test suite that uses the same parameters that you've been running on a cluster? Ideally it should run on the CI with a locally simulated cluster.
2. Remove any dependencies? The initial version should be able to run without building anything extra.

File: python/ray/experimental/raysort/main.py (hunk 1 above)
Comment (franklsf95) [reply python/ray/experimental/raysort/main.py:15]: I can make a dummy version of `sortlib` for now. I'll add a test for CI.

**Review by stephanie-wang — APPROVED** [review]
Thanks! You'll need to add the pytest to `python/ray/tests/BUILD` also. Is the test working now?

**Comment by franklsf95** [comment]
@stephanie-wang I added to `python/ray/tests/BUILD`. There was a lint error; now let's wait for the test to pass.

**Comment by franklsf95** [comment]
@stephanie-wang Test failures do not look relevant. The Python tests passed: https://buildkite.com/ray-project/ray-builders-pr/builds/7204#1561cdc6-84eb-4711-ba94-b70cbc28a391

**Comment by stephanie-wang** [comment]
Looks like the test is failing on windows. Want to just skip it there? You can check out the codebase for how to do that.

**Comment by franklsf95** [comment]
@stephanie-wang fixed!

**Comment by franklsf95** [comment]
Windows test also fixed.

**Pull request merged**